### ISSUE TITLE
Support SmoothL1Loss in Liger Kernel

### ISSUE THREAD

**Issue by yubofredwang** [linkedin/Liger-Kernel#817]
### 🚀 The feature, motivation and pitch

Smooth L1 loss is a regression loss function that combines the benefits of both L1 (Mean Absolute Error) and L2 (Mean Squared Error) loss. This loss function is commonly used in CNN models. One use case in LLM world is EAGLE draft model training combines this regression loss with the CE loss. See: https://github.com/SafeAILab/EAGLE/blob/a0e1e2c08604f0d215c531d9655b7ebacedde130/eagle/train/main.py#L231

Supporting this kernel can help speed up the draft model training and save memories considering the dimension of X and Y are BT, V(Batch, Seq, Vocab)

Speed Profilling vs torch native
================================================================================
Config          Reduction  PyTorch (ms)    Triton (ms)     Speedup   
------------------------------------------------------------------------------------------------------------------------
1024x131072     none       2.103           0.452           4.65      
2048x131072     none       4.123           0.823           5.01      
4096x131072     none       8.171           1.573           5.20      
8192x131072     none       16.300          3.073           5.31      
16384x131072    none       32.543          6.062           5.37      

Peak Memory Usage Profiling
==================================================
PyTorch Peak Memory: 168.01 MB
Triton Peak Memory:  72.00 MB
Memory Ratio:        2.33x

**Comment by yubofredwang** [comment]
PR In progress: https://github.com/linkedin/Liger-Kernel/pull/816
